core/overdrive.py

```diff
@@ -1267,22 +1268,34 @@ def book_info_to_metadata(
 
                 # Samples become links.
                 if "samples" in format:
-                    overdrive_name = format["id"]
-                    internal_names = list(cls.internal_formats(overdrive_name))
-                    if not internal_names:
-                        # Useless to us.
-                        continue
-                    for content_type, drm_scheme in internal_names:
-                        if Representation.is_media_type(content_type):
-                            for sample_info in format["samples"]:
-                                href = sample_info["url"]
+                    for sample_info in format["samples"]:
+                        href = sample_info["url"]
+                        # Have we already parsed this sample? Overdrive repeats samples per format
+                        if href in sample_hrefs:
+                            continue
+
+                        # Every sample has its own format type
+                        overdrive_format_name = sample_info.get("formatType")
+                        if not overdrive_format_name:
+                            # Malformed sample
+                            continue
+                        internal_names = list(
+                            cls.internal_formats(overdrive_format_name)
+                        )
+                        if not internal_names:
+                            # Useless to us.
+                            continue
+
+                        for content_type, drm_scheme in internal_names:
```

Comment:
It looks like we are throwing away the `drm_scheme` here. Is this is because all the samples don't have a `drm_scheme` set?

If `drm_scheme` isn't `None`, maybe we want to log a warning, and not create a representation so we don't end up with sample links that have DRM that we are not properly accounting for.